lib/SILOptimizer/Mandatory/MandatoryInlining.cpp

```diff
@@ -418,6 +419,27 @@ runOnFunctionRecursively(SILFunction *F, FullApplySite AI,
                          << " into @" << InnerAI.getFunction()->getName()
                          << "\n");
 
+      // If we intend to inline a thick function, then we need to balance the
+      // reference counts for correctness.
+      if (IsThick && I != ApplyBlock->begin()) {
+        // We need to find an appropriate location for our fix up code
+        // We used to do this after inlining Without any modifications
+        // This caused us to add a release in a wrong place:
+        // It would release a value *before* retaining it!
+        // It is really problematic to do this after inlining -
+        // Finding a valid insertion point is tricky:
+        // Inlining might add new basic blocks and/or remove the apply
+        // We want to add the fix up *just before* where the current apply is!
+        // Unfortunately, we *can’t* add the fix up code here:
+        // Inlining might fail for any reason -
```

Comment:
calling the inliner might fail / not inline the callee for various reasons (a call that can not be inlined for example). see its implementation. the pass bails out on that case. we only know if the inliner was successful *after the fact*